File: elgcsharp/PublicKey.cs
namespace sl.crypto.elgamal.keys
{
	using sl.crypto.elgamal.parameter;
	using sl.crypto.util;
	/**
	 * <summary>
 	 * Autor: Matthias Koch
 	 * Version: 0.5
 	 * 
 	 * Klasse welche ein konkretes Objekt für die Darstellung des öffentlichen Schlüssels
 	 * darstellt.
 	 * </summary>  
 	 */
	public class PublicKey : Key
	{
		private BigInteger y;
		//erlaube lesenden Zugriff
		public BigInteger Y
		{
			get{return y;} 
		}
		/**
		 * 
		 * <param name="y"/>
		 * <param name="params"/>
		 */
		public PublicKey(BigInteger y, Parameter parameter):base(false, parameter)
		{			
			this.y = y;
		}
		/**
		 * 
		 */
		public bool equals(object obj)
		{
			if (!(obj is PublicKey))
			{
				return false;
			}
			PublicKey pKey = (PublicKey) obj;
			if (!pKey.Y.Equals(y))
			{
				return false;
			}
			return Equals(obj);
		}
	}
}

File: elgjava/sl/crypto/elgamal/keys/PublicKey.java
package sl.crypto.elgamal.keys;

import java.math.BigInteger;

import sl.crypto.elgamal.parameter.Parameter;
/**
 *  Klasse welche ein konkretes Objekt für die Darstellung des öffentlichen Schlüssels
 * darstellt.
 * 
 * @author Matthias Koch
 * @version 0.5 
 */
public class PublicKey extends Key
{
	private BigInteger y;
	/**
	 * 
	 * @param y
	 * @param params
	 */
	public PublicKey(BigInteger y, Parameter params)
	{
		super(false, params);
		this.y = y;
	}
	/**
	 * 
	 * @return Biginteger y
	 */
	public BigInteger getY()
	{		
		return y;
	}
	/**
	 * 
	 */
	public boolean equals(Object obj)
	{
		if (!(obj instanceof PublicKey))
		{
			return false;
		}
		PublicKey pKey = (PublicKey) obj;
		if (!pKey.getY().equals(y))
		{
			return false;
		}
		return super.equals(obj);
	}
}
